### PR TITLE
Implement new libsolc API

### LINKED ISSUE #2864: libsolc and public API
I think with the JSON I/O specification we could consider our interface quite stable and could consider making it public (it is already public as an emscripten binary) in terms of a statically linked shared library on Linux similarly to the static build based on Alpine.

For that I'd propose to:
- rename `soljson` / `jsonCompiler` to `libsolc`
- create three public methods: `solcLicense`, `solcVersion`, `solcCompile` (aka. `license`, `version`, `compileStandard` currently)
- retire the obsolete method in the future (`license`, `version`, `compileJSON`, `compileJSONMulti`, `compileJSONCallback` and `compileStandard`) 
- build `libsolc` as a statically linked shared library so that the only symbols it exposes are those three and doesn't import anything
- include `libsolc` in `make install` and the releases

@chriseth @chfast what do you think?

### PR DESCRIPTION
**Pull request by axic**
Fixes #2864.

### CODE REVIEW

**Comment by axic** [comment]
This is backwards compatible, we could deprecate the old API later.

File: libsolc/libsolc.cpp [hunk 1]
```diff
@@ -299,4 +299,17 @@ extern char const* compileStandard(char const* _input, CStyleReadFileCallback _r
...
 }
+extern char const* solidity_license()
+{
+	return license();
+}
+extern char const* solidity_version()
+{
+	return version();
+}
+extern char const* solidity_compile(char const* _input, CStyleReadFileCallback _readCallback)
```
Comment (axic) [inline libsolc/libsolc.cpp]: @chfast could we just use `alias` here ?

Comment (chfast) [reply libsolc/libsolc.cpp]: There is no such thing as alias for function in C++. You can have a `constexpr` pointer to other function locally, but that is something different (will not be exported as a function).

1. `extern` keyword is wrong here.
2. You should move the implementation to `solidity_license()` and make the local alias `license()`. See example in ethash library: https://github.com/chfast/ethash/blob/master/include/ethash/ethash.hpp#L52-L53.

Comment (axic) [reply libsolc/libsolc.cpp]: Yeah that will be a next step, but for now this is fine.

**Comment by axic** [comment]
@chriseth any comments on this one? I have appropriate changes for solc-js, but github push is blocked by the wifi atm.

**Review by chriseth — DISMISSED** [review]

**Comment by axic** [comment]
Merging this now.

@chriseth this was approved, but had a compilation error.

**Pull request merged**